lwm2m-client-button-sensor.c

```diff
@@ -108,40 +103,24 @@ Options options =
  **************************************************************************************************/
 
 
-void ConstructObjectTree(Lwm2mContextType * context)
+void ConstructObjectTree(AwaStaticClient *client)
 {
-	Lwm2m_Debug("Construct object tree\n");
-
-	Lwm2m_RegisterSecurityObject(context);
-	if (options.BootStrap != NULL)
-	{
-		Lwm2m_PopulateSecurityObject(context, options.BootStrap);
-	}
-	Lwm2m_RegisterServerObject(context);
-	Lwm2m_RegisterDeviceObject(context);
-
-	Lwm2m_RegisterFlowObject(context);
-	Lwm2m_RegisterFlowAccessObject(context);
-
-	DigitalInput_RegisterDigitalInputObject(context);
-	DigitalInput_AddDigitialInput(context, 0);
-	DigitalInput_AddDigitialInput(context, 1);
+	DefineDeviceObject(client);
+	DefineFlowObject(client);
+	DefineFlowAccessObject(client);
+	DefineDigitalInputObject(client);
 }
 
-Lwm2mContextType * Lwm2mClient_Start()
+void AwaStaticClient_Start(AwaStaticClient *client)
 {
-	Lwm2m_SetOutput(stdout);
-	Lwm2m_SetLogLevel((options.Verbose) ? DebugLevel_Debug : DebugLevel_Info);
-	Lwm2m_PrintBanner();
-	Lwm2m_Info("LWM2M client - CoAP port %d\n", options.CoapPort);
-	Lwm2m_Info("LWM2M client - IPC port %d\n", options.IpcPort);
-
-	CoapInfo * coap = coap_Init("0.0.0.0", options.CoapPort,
-		(options.Verbose) ? DebugLevel_Debug : DebugLevel_Info);
-	Lwm2mContextType * context = Lwm2mCore_Init(coap, options.EndPointName);
-	ConstructObjectTree(context);
-
-	return context;
+	AwaStaticClient_SetLogLevel((options.Verbose) ? AwaLogLevel_Debug : AwaLogLevel_Warning);
+	printf("LWM2M client - CoAP port %d\n", options.CoapPort);
```

Comment:
There aren't any AwaLog methods in the apis so had to replace the lwm2m log methods by printfs